src/test/java/io/openliberty/tools/intellij/it/SingleModMPProjectTestCommon.java

```diff
@@ -68,6 +69,10 @@ public static void cleanup() {
         UIBotTestUtils.closeProjectView(remoteRobot);
         UIBotTestUtils.closeProjectFrame(remoteRobot);
         UIBotTestUtils.validateProjectFrameClosed(remoteRobot);
+
+        // Delete the Gradle and Maven project directories with spaces in their names that were created during the environment setup for tests.
+        deleteDirectoryIfExists(getGradleProjectSIDDirPath());
```

Comment:
Can we move these lines of codes to Gradle and Maven classes. 
Once the Gradle tests are completed, you can delete the Gradle project. Similarly, after the Maven project is finished, you can delete the Maven project as well.

Reply:
@aparnamichael , II initially tried the same approach, but the directory is still present after the corresponding Gradle and Maven tests are completed. When I debugged, I found that the cleanup() method in this class is executing last. It uses four methods (lines 68-71) that utilize the project directory path, which is causing the directory to remain after the tests are completed.

Reply:
My understanding is lines 68-71 is not using directory path, they are trying to find the UI element to close the project.  I moved the deleteDirectoryIfExists() to the GradleSingleModMPSIDProjectTest and MavenSingleModMPSIDProjectTest @AfterAll method. And the Test was working fine. Please try the same.

Reply:
I tried the same way you suggested, but the UI test is not completing all the actions. Specifically, the code below is not executing.

```
@AfterAll
public static void cleanup() {
    UIBotTestUtils.closeLibertyToolWindow(remoteRobot);
    UIBotTestUtils.closeProjectView(remoteRobot);
    UIBotTestUtils.closeProjectFrame(remoteRobot);
    UIBotTestUtils.validateProjectFrameClosed(remoteRobot);
}
```

The tests are stopping after executing the code we wrote in the GradleSingleModMPSIDProjectTest and MavenSingleModMPSIDProjectTest classes.

```
@AfterAll
public static void cleanup() {
    deleteDirectoryIfExists(PROJECTS_PATH_NEW);
}
```

Reply:
Can we change the code like this.

/**
 * Cleanup.
 */
@AfterAll
public static void cleanup() {
    closeProjectView();
}

protected static void closeProjectView() {
    // Parent cleanup logic
    UIBotTestUtils.closeLibertyToolWindow(remoteRobot);
    UIBotTestUtils.closeProjectView(remoteRobot);
    UIBotTestUtils.closeProjectFrame(remoteRobot);
    UIBotTestUtils.validateProjectFrameClosed(remoteRobot);
}

child class

@AfterAll
public static void cleanup() {
    closeProjectView();
    deleteDirectoryIfExists(PROJECTS_PATH_NEW);
}

Reply:
@aparnamichael , I have updated in the PR changes as per your suggestions.